### PR TITLE
Set checkout country to one from shipping address

### LINKED ISSUE #5780: Can not buy product if product unavailable for the default country
### What I'm trying to achieve
Buy product with is available in the shipping zone.

### Steps to reproduce the problem
1. Create a product with is unavailable in the shipping zone related to `DEFAULT_COUNTRY` and is available in another shipping zone. 
2. Try to create a checkout with an address in another shipping zone.

#### Example
Product is available in Europe, `DEFAULT_COUNTRY` = "US". 
If I try to buy this product with shipping to Poland I got the error:
`{"checkoutCreate": {"errors": [{"code": "INSUFFICIENT_STOCK", "field": "quantity", "message": "Could not add item AAATestEU (AAATestEU). Only 0 remaining in stock.", "__typename": "CheckoutError"}], "checkout": null, "__typename": "CheckoutCreate"}}}]`

### LINKED ISSUE #5781: Lack of availability validation on `checkoutShippingAddressUpdate`
### What I'm trying to achieve
Change checkout address. 

### Steps to reproduce the problem
1. Create product available only in the shipping zone related to `DEFAULT_COUNTRY` 
2. Create checkout for any address in `DEFAULT_COUNTRY` 
3. Execute `checkoutShippingAddressUpdate` with an address out of the shipping zone related to `DEFAULT_COUNTRY` 

### What I expected to happen
Validation shouldn't allow changing checkout shipping addresses if the product is unavailable for shipping zone related to a new address.

### PR DESCRIPTION
**Pull request by koradon**
This also checks if variants in checkouts are available in the shipping zone from shipping country

I want to merge this change because...
Fixes #5780
Fixes #5781 

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by koradon** [comment]
@maarcingebala Yes, I wanted to do #5781 in separate PR but it can be done in this one

**Comment by maarcingebala** [comment]
I didn't see that issue. It seems very related so I think it should be shipped in one PR.

File: saleor/graphql/checkout/mutations.py [hunk 1]
```diff
@@ -507,6 +514,22 @@ class Meta:
...
+        variants = list(
+            product_models.ProductVariant.objects.filter(
+                id__in=variant_ids
+            ).prefetch_related("product__product_type")
+        )
+        quantities = [line.quantity for line in lines]
+
+        check_lines_quantity(variants, quantities, country)
+
+        return variants, quantities
```
Comment (IKarbowiak) [inline saleor/graphql/checkout/mutations.py:531]: We not using it so probably can be removed.

File: saleor/graphql/checkout/tests/test_checkout.py [hunk 2]
```diff
@@ -527,6 +527,68 @@ def test_checkout_create_check_lines_quantity_multiple_warehouse(
     assert data["errors"][0]["field"] == "quantity"
 
 
+def test_checkout_create_sets_country_from_shipping_address_country(
```
Comment (fowczarek) [inline saleor/graphql/checkout/tests/test_checkout.py:530]: Maybe we should set `DEFAULT_COUNTRY`  for any other country than `US`?

File: saleor/graphql/checkout/tests/test_checkout.py [hunk 3]
```diff
@@ -527,6 +527,68 @@ def test_checkout_create_check_lines_quantity_multiple_warehouse(
...
+    assert not Checkout.objects.exists()
+    response = user_api_client.post_graphql(MUTATION_CHECKOUT_CREATE, variables)
+    content = get_graphql_content(response)
+    data = content["data"]["checkoutCreate"]
+    assert data["created"] is True
+    checkout = Checkout.objects.first()
+    assert checkout.country == "US"
+
+
+def test_checkout_create_check_lines_quantity_different_shipping_zone_insufficient_stocks(  # NOQA
```
Comment (fowczarek) [inline saleor/graphql/checkout/tests/test_checkout.py:558]: I think we should use the test name longer than the max line length. Maybe we can use a shorter name and add some comment if still need to add more information.

Comment (fowczarek) [reply saleor/graphql/checkout/tests/test_checkout.py:558]: If you set `DEFAULT_COUNTRY` in decorator before the test you can remove `different_shipping_zone` from the test name.

File: saleor/graphql/checkout/tests/test_checkout.py [hunk 4]
```diff
@@ -527,6 +527,68 @@ def test_checkout_create_check_lines_quantity_multiple_warehouse(
...
+    user_api_client,
+    variant_with_many_stocks_different_shipping_zones,
+    graphql_address_data,
+):
+    variant = variant_with_many_stocks_different_shipping_zones
+    stock = Stock.objects.filter(
+        warehouse__shipping_zones__countries__contains="US", product_variant=variant
+    ).first()
+    stock.quantity = 0
+    stock.save()
```
Comment (fowczarek) [inline saleor/graphql/checkout/tests/test_checkout.py:568]: Maby?
```suggestion
    stock = Stock.objects.filter(
        warehouse__shipping_zones__countries__contains="US", product_variant=variant
    ).update(quantity=0)
```

File: saleor/graphql/checkout/tests/test_checkout.py [hunk 5]
```diff
@@ -1278,6 +1340,92 @@ def test_checkout_shipping_address_update(
...
+):
+    variant = variant_with_many_stocks_different_shipping_zones
+    checkout = Checkout.objects.create()
+    checkout.set_country("PL", commit=True)
+    add_variant_to_checkout(checkout, variant, 1)
+    stock = Stock.objects.filter(
+        warehouse__shipping_zones__countries__contains="US", product_variant=variant
+    ).first()
+    stock.quantity = 0
+    stock.save()
```
Comment (fowczarek) [inline saleor/graphql/checkout/tests/test_checkout.py:1409]: Maby?
```suggestion
    stock = Stock.objects.filter(
        warehouse__shipping_zones__countries__contains="US", product_variant=variant
    ).update(quantity=0)
```

File: saleor/tests/fixtures.py [hunk 6]
```diff
@@ -495,6 +495,27 @@ def shipping_zone(db):  # pylint: disable=W0613
     return shipping_zone
 
 
+@pytest.fixture
+def shipping_zones(db):
```
Comment (fowczarek) [inline saleor/tests/fixtures.py:499]: Maby we can use `bulk_create`?

File: saleor/tests/fixtures.py [hunk 7]
```diff
@@ -859,6 +880,16 @@ def variant_with_many_stocks(variant, warehouses_with_shipping_zone):
     return variant
 
 
+@pytest.fixture
+def variant_with_many_stocks_different_shipping_zones(
+    variant, warehouses_with_different_shipping_zone
+):
+    warehouses = warehouses_with_different_shipping_zone
+    Stock.objects.create(warehouse=warehouses[0], product_variant=variant, quantity=4)
+    Stock.objects.create(warehouse=warehouses[1], product_variant=variant, quantity=3)
```
Comment (fowczarek) [inline saleor/tests/fixtures.py:889]: Maby we can use bulk_create?

File: saleor/tests/fixtures.py (hunk 6 above)
Comment (koradon) [reply saleor/tests/fixtures.py:499]: Should I update every fixture then? I was basing on existing ones.

File: saleor/graphql/checkout/tests/test_checkout.py (hunk 2 above)
Comment (koradon) [reply saleor/graphql/checkout/tests/test_checkout.py:530]: Yes but we want to test scenario where your selected country is other than one from shipping address. `DEFAULT_COUNTRY` id the one I will get from `request.user.country`?

File: saleor/tests/fixtures.py (hunk 6 above)
Comment (koradon) [reply saleor/tests/fixtures.py:499]: Done

File: saleor/graphql/checkout/tests/test_checkout.py (hunk 2 above)
Comment (koradon) [reply saleor/graphql/checkout/tests/test_checkout.py:532]: Done

**Pull request merged**